### ISSUE TITLE
Wrong Etag for S3 CopyObject on Multipart Source

### ISSUE THREAD

**Issue by sebbsn** [localstack/localstack#3729]
# Type of request: This is a ...

[X ] bug report
[ ] feature request

# Detailed description

I need to test cases of Copying multipart objects on S3 for which source and destination will have different chunk sizes.

Copying a non-multipart object to a multipart object works as expected with localstack.
Copying a multipart object to another multipart object where both have different chunk sizes also works as expected.

But when copying a multipart object to a non-multipart object, the multipart etag gets wrongly copied by localstack from the source to the destination. While it should represent an md5.

## Expected behavior

When an object with an etag of form `<combinedMd5>-<nrOfChunks>` gets copied using a normal s3 CopyObject operation (not multipart copy) it is expected that the destination object will have an etag of form `<md5>`. This has been confirmed using a real S3 setup. 

## Actual behavior

Localstack's behaviour is that for a source object with etag of form `<combinedMd5>-<nrOfChunks>`, the destination object will have the same etag, even though a normal CopyObject operation is performed.

# Steps to reproduce

used versions:
* Java 11
* Junit 5
* Java AWS Sdk version: 1.11.935
* localstack-utils version: 0.2.5
* localstack container (latest from dockerhub localstack/localstack as of 15.03.2021 - 16:00 UTC+1): ad38c920294f2bd5bf933c2a000c17e0647dd145ef9bb93d3875d9648c373bdf

## Command used to start LocalStack

```
import cloud.localstack.docker.LocalstackDockerExtension;
import cloud.localstack.docker.annotation.LocalstackDockerProperties;
import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import org.junit.jupiter.api.extension.ExtendWith;

@Inherited
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@ExtendWith(LocalstackDockerExtension.class)
@LocalstackDockerProperties(
        services = {"s3"},
        useSingleDockerContainer = true,
        imageTag = LocalstackS3Test.LOCALSTACK_IMAGE_VERSION,
        pullNewImage = true)
public @interface LocalstackS3Test {
    // latest version from dockerhub localstack/localstack as of 15.03.2021 - 16:00 UTC+1
    String LOCALSTACK_IMAGE_VERSION = "@sha256:ad38c920294f2bd5bf933c2a000c17e0647dd145ef9bb93d3875d9648c373bdf";
}
```

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

```
import cloud.localstack.awssdkv1.TestUtils;
import com.amazonaws.services.s3.AmazonS3;
import com.amazonaws.services.s3.model.CopyObjectRequest;
import com.amazonaws.services.s3.model.ObjectMetadata;
import com.amazonaws.services.s3.model.PutObjectRequest;
import com.amazonaws.services.s3.transfer.TransferManager;
import com.amazonaws.services.s3.transfer.TransferManagerBuilder;
import java.io.ByteArrayInputStream;
import org.apache.commons.codec.digest.DigestUtils;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

@LocalstackS3Test
public class LocalStackS3CopyEtagTest {

    private static final String BUCKET = "test-copy-etag-bucket";
    public static final int MIB = 1024 * 1024;

    private final AmazonS3 s3Client;

    public LocalStackS3CopyEtagTest() {
        s3Client = TestUtils.getClientS3();
        s3Client.createBucket(BUCKET);
    }

    @Test
    void testCopyMultipartSourceToNonMultipartDestination() throws InterruptedException {
        TransferManager uploadTm = TransferManagerBuilder.standard()
                .withMultipartUploadThreshold(5L * MIB)
                .withMinimumUploadPartSize(5L * MIB)
                .withS3Client(s3Client)
                .build();

        byte[] content = new byte[6 * MIB];

        String source = "src";
        String dest = "dst";

        ObjectMetadata newOm = new ObjectMetadata();
        newOm.setContentLength(content.length);

        String sourceEtag = uploadTm.upload(new PutObjectRequest(BUCKET, source, new ByteArrayInputStream(content), newOm))
                .waitForUploadResult().getETag();
        
        assertEquals( 2, sourceEtag.split("-").length, "Source etag should be of form <etag>-<nrOfChunks>");
        assertEquals(2, Integer.parseInt(sourceEtag.split("-")[1]), "Source etag should hint for 2 chunks");

        String destEtag = s3Client.copyObject(BUCKET, source, BUCKET, dest).getETag();
        
        assertEquals( 1, destEtag.split("-").length, "Destination etag should be of form <md5>");
        assertEquals(DigestUtils.md5Hex(content), destEtag, "Destination etag should match it's content's md5");
    }
}
```

**Comment by whummer** [comment]
Thanks for reporting @sebbsn . This should be fixed in #4002 - can you please pull the `latest` Docker image and give it another try? Please let us know if the problem persists. Thanks!

**Comment by sebbsn** [comment]
thanks, the mentioned test is working with the current latest version (sha256:cc4299a01ae27fc0eedd069f3d910006b3d45fffd00a1b9f9e8dfd63f863535e)
